base/loading.jl

```diff
@@ -508,7 +508,8 @@ function stale_cachefile(modpath, cachefile)
             return true # cache file was compiled from a different path
         end
         for (f,ftime) in files
-            if mtime(f) != ftime
+            # Issue #13606: compensate for Docker images rounding mtimes
+            if mtime(f) ∉ (ftime, floor(ftime))
```

Comment:
@sjkelly, for one thing, it would be && not ||, and for another thing, you're potentially doing the mtime() twice.